packages/a11y/src/useAccessibleForm.ts

```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react';
+import { IdAllocator } from '@leafygreen-ui/lib';
+
+const inputIdAllocator = IdAllocator.create('input');
+const labelIdAllocator = IdAllocator.create('label');
+
+/**
+ * Hook that returns two sets of props that accessibly associate a label and its respective input element
+ * @param id Describes the input element
+ */
+const useAccessibleForm = (idProp: string) => {
+  const inputId = useMemo(() => idProp ?? inputIdAllocator.generate(), [
+    idProp,
+  ]);
+  const labelId = useMemo(() => labelIdAllocator.generate(), []);
```

Comment:
Micro-optimization, but can't we simply do:

```js
const labelId = useMemo(labelIdAllocator.generate, [])
```

Reply:
we get a warning from eslint if we don't pass an inline function